sdk/core/Azure.Core/src/Shared/HttpMessageSanitizer.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace Azure.Core
+{
+    internal class HttpMessageSanitizer
+    {
+        private readonly bool _logAllHeaders;
+        private readonly bool _logFullQueries;
+        private readonly string[] _allowedQueryParameters;
+        private readonly string _redactedPlaceholder;
+        private readonly HashSet<string> _allowedHeaders;
+        private const string LogAllValue = "*";
+
+        public HttpMessageSanitizer(string[] allowedQueryParameters, string[] allowedHeaders, string redactedPlaceholder = "REDACTED")
```

Comment:
It's not truly public, sure, but should we still be doing some null checks here, or handling null as a value parameter value (treat it as empty, probably)? A lot of services will be calling into this, so doing some validation to avoid a NullReferenceException might be good.

Reply:
This could also translate below to just passing a null `allowedHeaders` or `allowedQueryParameters` instead of using a more expensive O(n) check for "*".

Reply:
"*" is coming from the ClientOptions object and cant' be replaced by null.